### PR TITLE
Remove ambiguity around username

### PR DESCRIPTION
**Pull request by johnnyaug**
1. Add description to id field in API.
2. Add description to user ID field in lakectl.
3. Add "username" field to Auth API and deprecate "name".
4. (Bonus) reorder flags on lakectl auth commands so that pagination flags are at the bottom.

### CODE REVIEW

File: docs/reference/commands.md [hunk 1]
```diff
@@ -865,7 +865,7 @@ lakectl auth users credentials create [flags]
 
 ```
   -h, --help        help for create
-      --id string   user identifier (default: current user)
+      --id string   Username (email for password-based users). Defaults to current user.
```
Comment (nopcoder) [inline docs/reference/commands.md:868]: There are 3 places in this PR that we started adding '.' at the end of the help line of the argument. we need to align.

File: pkg/auth/service.go [hunk 2]
```diff
@@ -1273,9 +1281,13 @@ func (a *APIAuthService) ListUsers(ctx context.Context, params *model.Pagination
 	results := resp.JSON200.Results
 	users := make([]*model.User, len(results))
 	for i, r := range results {
+		username := r.Username
+		if username == "" {
+			username = swag.StringValue(r.Name)
```
Comment (nopcoder) [inline pkg/auth/service.go:1286]: We would like to return the Name as Username even if it is empty, as we will make it required. Unless we plan on the next PR remove this requirement.

File: pkg/auth/service.go [hunk 3]
```diff
@@ -1439,9 +1451,13 @@ func (a *APIAuthService) ListGroupUsers(ctx context.Context, groupDisplayName st
 	members := make([]*model.User, len(resp.JSON200.Results))
 
 	for i, r := range resp.JSON200.Results {
+		username := r.Username
+		if username == "" {
+			username = swag.StringValue(r.Name)
```
Comment (nopcoder) [inline pkg/auth/service.go:1456]: We would like to return the Name as Username even if it is empty, as we will make it required

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/auth/service.go (hunk 2 above)
Comment (johnnyaug) [reply pkg/auth/service.go:1286]: The username is in the Auth service API definition - here is the lakeFS OpenAPI definition - no Name field.

**Review by nopcoder — APPROVED** [review]

**Pull request merged**